ansible_collections/arista/cvp/plugins/module_utils/change_tools.py

```diff
@@ -0,0 +1,657 @@
+        """
+
+        task = {}
+        task['action'] = {}
+        task['action']['name'] = action
+        task['action']['args'] = {}
+        task['action']['args']['values'] = {}
+        task['action']['args']['values']['DeviceID'] = deviceID
+        task['name'] = name
+
+        cardID = self.__genID__()
+        self.ChangeControl['change']['stages']['values'][cardID] = task
+        self.__attachThing(cardID, stage)
+
+        return None
+
+
+class CvChangeControlTools():
+    """
+    CvImageTools Class to manage Cloudvision Change Controls
+    """
+    def __init__(self, cv_connection, ansible_module: AnsibleModule = None, check_mode: bool = False):
+        self.__cv_client = cv_connection
+        self.__ansible = ansible_module
+        self.__check_mode = check_mode
+        # A list of tuples ("Change Control Name", "Change control ID")
+        self.__cc_index = []
+        self.cvp_version = self.__cv_client.api.get_cvp_info()['version']
+        self.apiversion = self.__cv_client.apiversion
+
+    def __index_cc__(self):
+        """
+        Index the known Change Controls, creating an internal list of tuples
+        ( Change Control Name, Change Control ID )
+
+        Parameters
+        ----------
+        None:
+            Provided data structure defining the Change
+
+        Returns
+        -------
+        None:
+
+        """
+        MODULE_LOGGER.debug('Indexing Change Controls')
+        self.__cc_index.clear()
+
+        for entry in self.change_controls['data']:
+            self.__cc_index.append((entry['result']['value']['change']['name'], entry['result']['value']['key']['id']))
+
+        return None
+
+    def _find_id_by_name(self, name):
+        """
+        Find the ID of a change control, by name
+
+        Parameters
+        ----------
+        name: str
+            The name of the CC
+
+        Returns
+        -------
+        cc_id: list
+            A list of matching change control IDs
+        """
+        cc_id = []
+        # cc_id = list(filter(lambda x: name in x, self.__cc_index))
+        for k, v in self.__cc_index:
+            if name in k:
+                cc_id.append(v)
+        MODULE_LOGGER.debug('%d changes found', len(cc_id))
+        return cc_id
+
+    def get_all_change_controls(self):
+        """
+        Get all change controls on CVP
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        None
+
+        """
+        cc_list = []
+        MODULE_LOGGER.debug('Collecting Change controls')
+
+        if self.apiversion < 3.0:
+            MODULE_LOGGER.debug('Trying legacy API call')
+            cc_list = self.__cv_client.api.get_change_controls()
+
+        else:
+            # Rewrite on cvprac > 1.0.7
+            MODULE_LOGGER.debug('Using resource API call')
+            cc_list = self.__cv_client.get('/api/resources/changecontrol/v1/ChangeControl/all')
```

Comment:
we could use [change_control_get_all() ](https://github.com/aristanetworks/cvprac/blob/develop/cvprac/cvp_api.py#L3257)

Reply:
It's only in the dev branch, I'll update to use those when cvprac has them available in a release

Reply:
oh right, forgot we didn't tag the new release yet 😬